### PR TITLE
Build Okio with Gradle

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: benchmarks/build.gradle [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apply plugin: 'org.jetbrains.kotlin.platform.jvm'
+
+apply plugin: "me.champeau.gradle.jmh"
+
+sourceCompatibility = JavaVersion.VERSION_1_7
+targetCompatibility = JavaVersion.VERSION_1_7
```
Comment (JakeWharton) [inline benchmarks/build.gradle]: Could be 8 if we want...

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  dependencies {
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${versions.kotlin}"
+    classpath "me.champeau.gradle:jmh-gradle-plugin:0.4.5"
+  }
+
+  repositories {
+    mavenCentral()
+    jcenter()
+    maven { url "https://plugins.gradle.org/m2/" }
```
Comment (JakeWharton) [inline build.gradle]: there's `gradlePluginPortal()` now!

File: okio-kotlin/build.gradle [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'org.jetbrains.kotlin.platform.jvm'
+
+sourceCompatibility = JavaVersion.VERSION_1_6
+targetCompatibility = JavaVersion.VERSION_1_6
+
+dependencies {
+  implementation project(':okio')
```
Comment (JakeWharton) [inline okio-kotlin/build.gradle]: needs to be `api`

File: okio-kotlin/build.gradle [hunk 4]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'org.jetbrains.kotlin.platform.jvm'
+
+sourceCompatibility = JavaVersion.VERSION_1_6
```
Comment (digitalbuddha) [inline okio-kotlin/build.gradle]: some modules are 1.6, while others are 1.7 is that intentional?

File: okio/build.gradle [hunk 5]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'org.jetbrains.kotlin.platform.jvm'
+
+sourceCompatibility = JavaVersion.VERSION_1_7
+targetCompatibility = JavaVersion.VERSION_1_7
+
+dependencies {
+  implementation deps.kotlin.stdLib.jre7
```
Comment (NightlyNexus) [inline okio/build.gradle]: ditto Mike's comment: jre6?

Comment (Egorand) [reply okio/build.gradle]: We should probably move everything to jre7 (which is actually jdk7 now)

Comment (JakeWharton) [reply okio/build.gradle]: do we use anything from 7? i wouldn't bother unless we do

Comment (Egorand) [reply okio/build.gradle]: Diamond operator, underscores in numeric literals

Comment (JakeWharton) [reply okio/build.gradle]: That is not tied to this dependency

**Comment by naturalwarren** [comment]
Out of curiosity, what's the motivation behind this change?

**Comment by swankjesse** [comment]
@naturalwarren https://github.com/square/okio/issues/370

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      'jsr305': '3.0.2',
+      'animalSniffer': '1.10',
+      'junit': '4.12',
+      'assertj': '1.7.0'
+  ]
+
+  ext.deps = [
+      'kotlin': [
+          'stdLib': [
+              'common': "org.jetbrains.kotlin:kotlin-stdlib-common:${versions.kotlin}",
```
Comment (tbroyer) [inline build.gradle]: You don't need to specify the version, it's inherited from the plugin: https://kotlinlang.org/docs/reference/using-gradle.html#configuring-dependencies
> Starting with Kotlin 1.1.2, the dependencies with group `org.jetbrains.kotlin` are by default resolved with the version taken from the applied plugin.

File: benchmarks/build.gradle [hunk 7]
```diff
@@ -0,0 +1,15 @@
+apply plugin: 'org.jetbrains.kotlin.platform.jvm'
```
Comment (tbroyer) [inline benchmarks/build.gradle]: Shouldn't this just be `org.jetbrains.kotlin.jvm`? (without the `platform` part)

Comment (Egorand) [reply benchmarks/build.gradle]: You're right, it should be for this module

File: build.gradle [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+          ],
+          'test': [
+              'common': "org.jetbrains.kotlin:kotlin-test-common:${versions.kotlin}",
+              'annotations': "org.jetbrains.kotlin:kotlin-test-annotations-common:${versions.kotlin}",
+              'jre': "org.jetbrains.kotlin:kotlin-test-junit:${versions.kotlin}",
+          ],
+      ],
+      'jmh': "org.openjdk.jmh:jmh-core:${versions.jmh}",
+      'jsr305': "com.google.code.findbugs:jsr305:${versions.jsr305}",
+      'animalSniffer': "org.codehaus.mojo:animal-sniffer-annotations:${versions.animalSniffer}",
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply build.gradle]: I think this would be a great PR for a follow-up

File: build.gradle [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      'test': [
+          'junit': "junit:junit:${versions.junit}",
+          'assertj': "org.assertj:assertj-core:${versions.assertj}"
+      ]
+  ]
+
+  dependencies {
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:${versions.kotlin}"
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply build.gradle]: Is there any advantage to this? It's weirder than Gradle normal level of weird.

File: build.gradle [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      'animalSniffer': '1.10',
+      'junit': '4.12',
+      'assertj': '1.7.0'
+  ]
+
+  ext.deps = [
+      'kotlin': [
+          'stdLib': [
+              'common': "org.jetbrains.kotlin:kotlin-stdlib-common:${versions.kotlin}",
+              'jre6': "org.jetbrains.kotlin:kotlin-stdlib:${versions.kotlin}",
```
[the remark this answers is not in the document]
Comment (JakeWharton) [reply build.gradle]: There will eventually be 7 and 8 versions to support stdlib-jdk7 and stdlib-jdk8

File: .travis.yml [hunk 11]
```diff
@@ -8,6 +8,9 @@ addons:
     packages:
       - oracle-java8-installer # Updates JDK 8 to the latest available.
 
+script:
+  - ./gradlew clean build jmh
```
Comment (JakeWharton) [inline .travis.yml]: Does this run jmh or just build it? We shouldn't need to run it.

File: build.gradle [hunk 12]
```diff
@@ -1,44 +1,59 @@
+group = 'com.squareup.okio'
+version = '1.15.0-SNAPSHOT'
```
Comment (JakeWharton) [inline build.gradle]: I think both of these need to be in `allprojects { }` block.

File: okio/build.gradle [hunk 13]
```diff
@@ -1,12 +1,19 @@
+group = 'com.squareup.okio'
```
Comment (JakeWharton) [inline okio/build.gradle]: With the `allprojects { }` block you can get rid of this from the subproject.

**Comment by Egorand** [comment]
Alright, that should be it except for publishing setup, I'll do this as a follow-up, then we can get rid of Maven files

File: okio/build.gradle [hunk 14]
```diff
@@ -0,0 +1,29 @@
+description = 'Okio'
+
+apply plugin: 'java-library'
+apply plugin: 'osgi'
```
Comment (JakeWharton) [inline okio/build.gradle]: Oh dear...

Comment (Egorand) [reply okio/build.gradle]: Yeah, that's used for `Automatic-Module-Name`, I'll if there's a better way

Comment (tbroyer) [reply okio/build.gradle]: You don't actually need it to configure the jar manifest: https://docs.gradle.org/current/userguide/java_plugin.html#sub:manifest

**Pull request merged**